OasisDEX/oasis-earn-sc
# feat: prepare Morpho test env in Tenderly fork

## Ticket URL

https://app.shortcut.com/oazo-apps/story/12960/test-prepare-testing-environment-on-fork

## Description of Changes

Please list the changes introduced by this PR:
- Fixed Tenderly config to be able to verify deployed contracts automatically
- Added new `.env` variable to enable Tenderly automatic verification `TENDERLY_AUTOMATIC_VERIFICATION` that can be set to `true` or `false`
- Added `yarn deploy:morphoblue:integration` command to `dma-contracts` to deploy Morpho. It only works with `--network tenderly` for now
- Separated internal test Morpho deployment from deployment for a mainnet fork, so it could be reused to deploy to Tenderly
- Added Chainlink wrappers to act as Morpho oracles to get real prices on-chain
- Added some minor optimizations to helper classes

## How to Test

- Install Tenderly command line tool
- Login into Tenderly with the command line tool
- Create a fork in Tenderly
- Configure the Tenderly env variables in `.env`
```
## Tenderly config
TENDERLY_PROJECT=oasis
TENDERLY_FORK_URL=URL of the fork
TENDERLY_FORK_CHAIN_ID=Chain ID of the fork
TENDERLY_AUTOMATIC_VERIFICATION=true or false
```
- Fund the deployer account in Tenderly
- Go to `packages/dma-contracts` and run ``yarn deploy:morphoblue:integration --network tenderly`
- Check that the system has been deployed to the fork and that the contracts have been verified

## PR Definition of Done

- [✔️] All checks are passing
- [✔️] PR is linked to a corresponding ticket
- [✔️] PR title is clear and concise
- [✔️] Code has been self-reviewed and any fixes or improvements noted (See Code review standards in Notion)
- [✔️] Documentation has been updated if necessary
